### PR TITLE
[WIP] Payconiq implementation

### PR DESCRIPTION
**Pull request by JustNireon**
New way to pay for activities / mongoose. Still needs a lot of work. furthermore the PayconiqTransaction model has a lot of overlapping code with IdealTransaction. I will look into how I could clean this up / make it more efficient.

To Do:
- [x]  2 tab option for payconiq/Ideal
- [x]  app2app linking 
- [x]  checkmark if you should be able to pay for the activity (possibility to pay for activities beforehand)
- [x] more logging so treasurer can easily keep overview
- [x] possibly make a parent transaction model (how i currently do it there are 4 "transaction" models
- [x] Signature verfication 
- [x] Documentation on how to test the webhook.
- [ ] Magnificent-Sloth api endpoint to request qrcode / websocket connection to confirm transaction (probably should be another pr)
- [x] clean up translations

### CODE REVIEW

File: db/migrate/20200527123642_ideal_transactions_to_payment.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def change
+    rename_table :ideal_transactions, :payments
+    rename_column :payments, :redirect_uri, :ideal_redirect_uri
+
+    remove_column :payments, :transaction_type, :string
+
+    add_column :payments, :transaction_type, :integer, default: 0
+    add_column :payments, :payment_type, :integer, default: 0
+
+    change_column :checkout_transactions, :payment_method, :integer
```
Comment (Riscky) [inline db/migrate/20200527123642_ideal_transactions_to_payment.rb:11]: this doesn't work, just change the limit of payment_method and remove payment_type

**Comment by Riscky** [comment]
create markdown with testing instructions

File: app/models/payment.rb [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      # Payconiq payments
+    else
+      http = ConstipatedKoala::Request.new ENV['PAYCONIQ_DOMAIN']
+      self.token = Digest::SHA256.hexdigest("#{ member.id }#{ Time.now.to_f }")
+
+      request = http.post("/#{ ENV['PAYCONIQ_VERSION'] }/payments")
+
+      request.body = { :amount => (amount * 100).to_i,
+                       :currency => 'EUR',
+                       :callbackUrl => "http://75a349d5c728.ngrok.io/api/hook/payconiq" }.to_json
```
Comment (Riscky) [inline app/models/payment.rb:65]: make this and enviroment variable pls

**Comment by Riscky** [comment]
totals are not working

File: package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "intl-tel-input": "^16.0.5",
     "jquery": "^3.2.1",
     "jquery-ujs": "^1.2.2",
-    "sweetalert": "^1.1.3",
+    "sweetalert2": "^9.10.12",
```
Comment (Riscky) [inline package.json:17]: move this to a different PR pls

File: app/models/payment.rb (hunk 2 above)
Comment (JustNireon) [reply app/models/payment.rb]: Fixed

File: app/controllers/admin/apps_controller.rb [hunk 4]
```diff
@@ -13,8 +13,26 @@ def checkout
     @products = CheckoutProduct.where(:active => true).count
   end
 
-  def ideal
-    @transactions = IdealTransaction.order(created_at: :desc)
-                                    .paginate(page: params[:page], per_page: params[:limit] ||= 20)
+  def payments
+    ## This feels wack tbh somebody please come up with a better way.
+    @transactions = if params[:search].blank?
```
Comment (Riscky) [inline app/controllers/admin/apps_controller.rb:18]: can't you just build up your query per parameter?
so you get 1 if-statement per parameter

File: app/controllers/admin/home_controller.rb [hunk 5]
```diff
@@ -7,12 +7,20 @@ def index
...
 
     @transactions = CheckoutTransaction.count(:all)
+
     @recent = CheckoutTransaction.where("created_at >= ?", Time.zone.now.beginning_of_day).order(created_at: :desc).take(12)
+    @recentparticipants = Payment.where("created_at >= ?", Time.zone.now.beginning_of_day).where(transaction_type: :activity).order(created_at: :desc).take(12)
 
-    @unpayed = Participant.where(:reservist => false, :paid => false).joins(:activity).where('activities.start_date < NOW()').sum(:price) + Participant.where(:reservist => false, :paid => false, :price => nil).joins(:activity).where('activities.start_date < NOW() AND activities.price IS NOT NULL').sum('activities.price')
+    @unpayed = Participant.where(:reservist => false, :paid => false, :price => nil)
+                          .joins(:activity)
+                          .where('activities.is_payable IS TRUE AND activities.price IS NOT NULL')
```
Comment (Riscky) [inline app/controllers/admin/home_controller.rb:16]: Isn't there a ruby method for AND and NOT NULL?

File: app/controllers/api/webhook_controller.rb [hunk 6]
```diff
@@ -13,7 +13,16 @@ def mollie_redirect
...
   def mollie_hook
-    transaction = IdealTransaction.find_by_trxid!(params[:id])
+    transaction = Payment.find_by_trxid!(params[:id])
+    transaction.finalize! if transaction.update!
+
+    head :ok
+  end
+
+  def payconiq_hook
+    puts "hoook"
```
Comment (Riscky) [inline app/controllers/api/webhook_controller.rb:23]: don't forget to remove debug code

File: app/controllers/admin/home_controller.rb (hunk 5 above)
Comment (JustNireon) [reply app/controllers/admin/home_controller.rb:16]: This would change the query to a much uglier query afaik there is no real way to do this cleanly
